### PR TITLE
Refactor shelling out in Python

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
There's a lot of duplicated code in the Python ecosystem that deals with shelling out to package managers. This PR refactors it to always use the same helper.

This fixes most of #6110.

### CODE REVIEW

File: python/lib/dependabot/python/file_updater/poetry_file_updater.rb [hunk 1]
```diff
@@ -204,7 +203,7 @@ def run_poetry_update_command
         end
 
         def run_poetry_command(command, fingerprint: nil)
-          Helpers.run_poetry_command(command, fingerprint: fingerprint)
+          SharedHelpers.run_shell_command(command, fingerprint: fingerprint)
```
Comment (deivid-rodriguez) [inline python/lib/dependabot/python/file_updater/poetry_file_updater.rb:206]: Same as the other comment, but reversed. This used to raise an error including only stderr, now we'll raise an error include stdout and stderr concatenated.

Seems fine to I guess!

File: python/lib/dependabot/python/update_checker/poetry_version_resolver.rb [hunk 2]
```diff
@@ -309,7 +308,7 @@ def lockfile
         end
 
         def run_poetry_command(command, fingerprint: nil)
-          Helpers.run_poetry_command(command, fingerprint: fingerprint)
+          SharedHelpers.run_shell_command(command, fingerprint: fingerprint)
```
Comment (deivid-rodriguez) [inline python/lib/dependabot/python/update_checker/poetry_version_resolver.rb:311]: Same as the other comment in the other `run_poetry_command` method.

**Review by yeikel — COMMENTED** [review]
It's always great to see these cleanups and more so when the tests are happy 🥇

**Review by Nishnha — APPROVED** [review]
This LGTM. I would maybe get another approval from the team though since I haven't used `time_taken` in my debugging, but maybe someone else has found it useful and doesn't want it removed?

**Comment by deivid-rodriguez** [comment]
@Nishnha Not that this is not removing `time_taken`, the same code is present in the common helpers which is now used consistently:

https://github.com/dependabot/dependabot-core/blob/1b3368e3d7e4eecb5ccb9475151ddd09018f5221/common/lib/dependabot/shared_helpers.rb#L317-L348

Only changes in behavior should be the ones I noticed as inline comments in the PR.

**Pull request merged**